artemis-server/src/main/java/org/apache/activemq/artemis/core/protocol/core/impl/wireformat/ReplicationLargeMessageBeginMessage.java

```diff
@@ -44,11 +46,13 @@ public int expectedEncodeSize() {
    @Override
    public void encodeRest(final ActiveMQBuffer buffer) {
       buffer.writeLong(messageId);
```

Comment:
I'm not sure how this is possibly working...


expectedEncodeSize is not taking this new long into consideration...


Besides this will cause versioning issues...


So, you will need to do something like this on reading... 

On reading.. you will need to verify the positioning of the buffer.. against the declared size.. if there's another long coming.. otherwise it wouldn't read it.. and use the current semantic.


this is a -1.. can you fix it?